### PR TITLE
PE-814: 2022-01-21 Executive

### PR DESCRIPTION
**Pull request by godsflaw**
# Description

2022-01-21 Executive

# Contribution Checklist

- [x] PR title starts with `(PE-<TICKET_NUMBER>)`
- [x] Code approved
- [x] Tests approved
- [ ] CI Tests pass

# Checklist

- [x] Every contract variable/method declared as public/external private/internal
- [x] Consider if this PR needs the `officeHours` modifier
- [x] Verify expiration (`4 days + 2 hours` monthly and `30 days` for the rest)
- [x] Verify hash in the description matches [here](https://emn178.github.io/online-tools/keccak_256.html)
- [x] Validate all addresses used are in Goerli changelog or known
- [x] Notify any external teams affected by the spell so they have the opportunity to review
- [x] Deploy spell to Goerli `ETH_GAS="XXX" ETH_GAS_PRICE="YYY" make deploy`
- [x] Ensure contract is verified on `Goerli` etherscan
- [x] Change test to use Goerli spell address and deploy timestamp
- [ ] Run `make name="YYYY-MM-DD-archive-directory-name" archive-spell` to make an archive directory and copy `Goerli-DssSpell.sol`, `Goerli-DssSpell.t.sol`, and `Goerli-DssSpell.t.base.sol` 
- [ ] `squash and merge` this PR

### CODE REVIEW

**Comment by talbaneth** [comment]
LGTM for deployment.

verified:
-rates numbers.
-parameters match the sheet and polls (apart from what we discussed)
-parameters ok
-GUSD-A reduction ok since without auto-line.
-office hours ok (off since goerli)
-test cover and match the sheet.

tests pass:
compiling...
success.
Running 13 tests for DssSpellTest.json:DssSpellTest
[PASS] testCastCost() (gas: 732121)
[PASS] testCollateralIntegrations() (gas: 1767199)
[PASS] testFailTooEarly() (gas: 4587)
[PASS] testFailTooLate() (gas: 4553)
[PASS] testFailWrongDay() (gas: 4554)
[PASS] testFail_notScheduled() (gas: 7811)
[PASS] testOnTime() (gas: 730386)
[PASS] testSpellIsCast_GENERAL() (gas: 8235068)
[PASS] test_auth() (gas: 2645957)
[PASS] test_auth_in_sources() (gas: 3420739)
[PASS] test_bytecode_matches() (gas: 1638729)
[PASS] test_nextCastTime() (gas: 348013)
[PASS] test_use_eta() (gas: 346594)

File: template/Goerli-DssSpell.sol [hunk 1]
```diff
@@ -0,0 +1,58 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+//
+// Copyright (C) 2021 Dai Foundation
```
Comment (iamchrissmith) [inline template/Goerli-DssSpell.sol:3]: do we need to add 2022 here?

**Review by iamchrissmith — APPROVED** [review]
lgtm

File: template/Goerli-DssSpell.sol (hunk 1 above)
Comment (brianmcmichael) [reply template/Goerli-DssSpell.sol:3]: done

**Review by talbaneth — APPROVED** [review]
LGTM

**Comment by iamchrissmith** [comment]
post deploy tests pass:

```haskell
dapp-test: rpc block: latest
Running 12 tests for src/Goerli-DssSpell.t.sol:DssSpellTest
[PASS] testFail_notScheduled() (gas: 12325)
[PASS] test_auth_in_sources() (gas: 3327300)
[PASS] test_use_eta() (gas: 338182)
[PASS] testOnTime() (gas: 657740)
[PASS] test_nextCastTime() (gas: 339182)
[PASS] testFailTooEarly() (gas: 8833)
[PASS] test_bytecode_matches() (gas: 1106933)
[PASS] test_auth() (gas: 2518845)
[PASS] testSpellIsCast_GENERAL() (gas: 8647604)
[PASS] testFailWrongDay() (gas: 9053)
[PASS] testCastCost() (gas: 659634)
[PASS] testFailTooLate() (gas: 9207)
```

Block number: 6235287

**Comment by brianmcmichael** [comment]
I updated the archive spell script because it was broken. I used the naming convention from spells-mainnet and updated the Makefile so we don't need to pass an arg. Please review before we merge.

**Comment by iamchrissmith** [comment]
> I updated the archive spell script because it was broken. I used the naming convention from spells-mainnet and updated the Makefile so we don't need to pass an arg. Please review before we merge.

I like these changes 👍

File: Makefile [hunk 2]
```diff
@@ -7,4 +7,4 @@ test-dev        :; ./test-dssspell.sh match="$(match)" optimizer="0"
 test-forge      :; ./test-dssspell-forge.sh match="$(match)" block="$(block)"
 deploy          :; make && dapp create DssSpell | xargs ./verify.py DssSpell
 flatten         :; hevm flatten --source-file "src/Goerli-DssSpell.sol" > out/flat.sol
-archive-spell   :; ./archive-dssspell.sh $(name)
+archive-spell   :; ./archive-dssspell.sh "$(if $(date),$(date),$(shell date +'%Y-%m-%d'))"
```
Comment (talbaneth) [inline Makefile:10]: This is really confusing as both the expected env var is called "date" and the shell cmd is called "date".
The general standard is to have env vars capitalized (which will requie changing also" match" and "block" above).
I don't mind if we leave it as is for now and change it later.

Comment (talbaneth) [reply Makefile:10]: Ah nvm, seeing now that since it's a makefile you pass date=X so I guess it doesn't have to be an env var.
So You can ignore it.

**Review by talbaneth — APPROVED** [review]
LGTM to merge.

**Pull request merged**